sound/soc/sof/ipc4-topology.c

```diff
@@ -1083,13 +1088,43 @@ static int sof_ipc4_init_output_audio_fmt(struct snd_sof_dev *sdev,
 		return 0;
 	}
 
+	/* if module has no output_pin_binding specified in topology, use pin 0 output formats */
```

Comment:
Why pin 0? If there is no output_pin_binding specified, shouldn't we choose the format that matches the reference parms?

Reply:
@bardliao if there's no output pin binding in the topology, the module likely has identical audio formats for all pins or only has 1 output pin and in both bcases, the audio format definitions in topology will ignore setting the pin_index and by default the pin index will be 0

Reply:
But if the topology doesn't specify output pin and each pin can support different formats, we will only use pin 0's format even the widget may not bind to pin 0 with this change.